base_optional_create_edit/models/ir_ui_view.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2024 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from lxml import etree
+
+from odoo import api, models
+
+
+class Model(models.AbstractModel):
+    _inherit = "base"
+
+    @api.model
+    def get_views(self, views, options=None):
+        res = super().get_views(views, options)
+        views = res.get("views")
+        view_form = views["form"]["arch"]
```

Comment:
What's happening if you're loading views without form view?
You should be more broad for this generic code:
```
view_form = res.get("views", {}).get("form", {}).get("arch")
if view_form:
    tree = ...
    ...
```
Because if someone is not fit as you expect (but still correct of course), everything will be broken.